### DISCUSSION TITLE
Image Generation Button Missing After Update to Docker image v0.4.2

### DISCUSSION THREAD

**Discussion by br00t4c** [open-webui/open-webui#7131] [Issues]
## Installation Method

Docker

## Environment

- **Open WebUI Version:** 0.4.2
- **Ollama (if applicable):** 0.4.2

- **Operating System:** Ubuntu 24.04
- **Browser (if applicable):** Firefox 132.0.2 (64-bit)

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

Click the image generation button at the bottom of a response in order to generate an image.

## Actual Behavior:

Image generation button is missing. Image generation config (tried both AUTOMATIC111 and ComfyUI) is correct and image generation endpoint is reachable, but no image generation button is shown in chat responses.

## Description

**Bug Summary:**

Image generation button is missing from chat responses.

## Reproduction Details

**Steps to Reproduce:**

Upgraded to v0.4.2 of the Docker image

## Logs and Screenshots

**Browser Console Logs:**

N/A

**Docker Container Logs:**

N/A

**Screenshots/Screen Recordings (if applicable):**

![Screenshot from 2024-11-20 18-11-29](https://github.com/user-attachments/assets/af5207e3-6fac-4a3a-b88a-d8d55dc1c755)

## Additional Information

N/A

**Comment by tjbck** [comment]
<img width="468" alt="image" src="https://github.com/user-attachments/assets/34bd608f-578c-4867-aae1-aef096bf4a3e">

Unable to reproduce, keep us updated!

**Comment by atnjqt** [comment]
I'm running into the same thing as of like v0.5.4, I've moved to v0.5.6 today and still no image generation!

**Reply by br00t4c** [reply]
Deleting the sqlite db and restoring just chats to a clean container resolved the issue for me in the end. Took the lazy way out.

**Reply by atnjqt** [reply]
by any chance did you run environment variables when executing the new container? wondering if `SAFE_MODE` is at all related

**Reply by br00t4c** [reply]
No, One additional note: I noticed that in v0.5.5 they introduced a feature "🖼️ Toggle Image Generation On/Off: In the chat input menu, you can now easily toggle image generation before initiating chats, providing greater control and flexibility to suit your needs." ... perhaps check that image generation is not toggled to "off" before nuking your db?

**Reply by atnjqt** [reply]
huh yeah great point -- as a consumer of open source software, the least I can (should) do is read the CHANGELOG 😅 -- with that said, I have no clue what the 'chat input menu' is!

**Reply by br00t4c** [reply]
I *think* maybe this???
![image](https://github.com/user-attachments/assets/b893c3de-4513-4872-a6b0-674559ad0409)

**Reply by atnjqt** [reply]
I cannot for the life of me figure out where these options are in my environment! when I click that `+` icon all I see is _Capture_ and _Uploads Files_ 
![image](https://github.com/user-attachments/assets/898c0f53-3cbc-42d5-87e3-301fbf9652e7)

**Reply by atnjqt** [reply]
Ohh huh maybe it's a tool and I wasn't aware! https://openwebui.com/t/justinrahb/image_gen

**Comment by nachogmd** [comment]
![imagen](https://github.com/user-attachments/assets/1ea78f49-3009-45c2-bac4-1414000f2876)

First of all you need to enable image generation. And you either have an api key for existing service or you'll need to install https://github.com/AUTOMATIC1111/stable-diffusion-webui and configure appropriately in OpenWebUI

**Comment by atnjqt** [comment]
I've almost got it:
- Following instructions for Tools here: https://docs.openwebui.com/features/plugin/tools/#how-can-i-use-tools
- Installed the following https://openwebui.com/t/justinrahb/image_gen
- and of course I've enabled the image generation in admin settings, trying a myriad of different combos of dalle-2 and dalle-3 per API specifics here https://platform.openai.com/docs/guides/images#generations
- and enabled for models, but when I attempt the image doesn't actually generate properly

<img width="1117" alt="image" src="https://github.com/user-attachments/assets/7c696233-a8d8-4702-a603-5e6d92d6737f" />

<img width="1089" alt="image" src="https://github.com/user-attachments/assets/9322b8fb-11c3-4d4a-aedb-025dd985aeb3" />

- what is so strange, is that ONE image did actually generate but on rerun that same command in a different chat it overwrote the cat image with an entirely new cat pic
<img width="1127" alt="image" src="https://github.com/user-attachments/assets/d0c3df63-9eaf-42d8-93b3-ee377428e741" />

**Reply by atnjqt** [reply]
if it's helpful, I'm running the docker OpenAI only installation: https://github.com/open-webui/open-webui?tab=readme-ov-file#installation-for-openai-api-usage-only

**Comment by atnjqt** [comment]
Alright so I've finally figured out how to get image generation working using the **Tools** implementation. I had to pin my container image to `v0.5.3` as the tool does suggest is the minimum supported version
![image](https://github.com/user-attachments/assets/1d523bc0-ad8b-4ce0-9edc-61c1fab71805)

**Comment by atnjqt** [comment]
Alright so digging into things, there were some changes in the `backend/open_webui/routers/images.py` which have impacted that community driven tool.

specifically the diff across `async def image_generations` in how images are base64 json dumped

- https://github.com/open-webui/open-webui/blob/v0.5.3/backend/open_webui/routers/images.py#L491-L499
- and the latest https://github.com/open-webui/open-webui/blob/main/backend/open_webui/routers/images.py#L480-L484

**Reply by atnjqt** [reply]
@justinh-rahb thanks for this Image Gen tool btw! probably just some minor changes in how the images are received

**Reply by atnjqt** [reply]
I think the git blame here helps outline this https://github.com/open-webui/open-webui/commit/159578dfd41a5fc68ac6c7cfdcf0db837600fe35
